source/Calamari/Integration/Scripting/FSharp/Bootstrap.fsx

```diff
@@ -0,0 +1,68 @@
+﻿module Octopus
+
+open System
+open System.Collections.Generic
+open System.IO
+open System.Text
+open System.Net
+open System.Security.Cryptography
+
+let private encode (value:string) = System.Text.Encoding.UTF8.GetBytes(value) |> Convert.ToBase64String
+let private decode (value:string) = Convert.FromBase64String(value) |> System.Text.Encoding.UTF8.GetString
+
+let private writeServiceMessage name content = 
+    printfn "##octopus[%s %s]" name content
+
+let tryFindVariable name =
+    match name |> encode with
+{{VariableDeclarations}}
+
+let findVariable name =
+    match name |> tryFindVariable with
+    | Some x -> x
+    | None -> raise (System.Collections.Generic.KeyNotFoundException(name + " variable has not been found."))   
+
+let findVariableOrDefault defaultValue name =
+    match name |> tryFindVariable with
+    | Some x -> x
+    | None -> defaultValue       
+
+let decryptString encrypted iv =
+    let key =  fsi.CommandLineArgs.[fsi.CommandLineArgs.Length - 1]
+    use algorithm = new AesCryptoServiceProvider(Mode = CipherMode.CBC, Padding = PaddingMode.PKCS7, KeySize = 128, BlockSize = 128, Key = Convert.FromBase64String(key), IV =  Convert.FromBase64String(iv))
+    use decryptor = algorithm.CreateDecryptor()
+    use memoryStream = new MemoryStream(Convert.FromBase64String(encrypted))
+    use cryptoStream = new CryptoStream(memoryStream, decryptor, CryptoStreamMode.Read)
+    use streamReader = new StreamReader(cryptoStream, Encoding.UTF8)
+    streamReader.ReadToEnd();
+
+let initializeProxy () =
```

Comment:
If you're merging into develop for 3.4 you'll need to expand this a little to allow for custom proxy hosts, see https://github.com/OctopusDeploy/Calamari/blob/develop/source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1